src/NuGetPackageVerifier/Rules/AssemblyHasCommitHashAttributeRule.cs

```diff
@@ -11,18 +11,20 @@
 
 namespace NuGetPackageVerifier.Rules
 {
-    public class AssemblyHasCommitHashAttributeRule : AssemblyHasAttributeRuleBase
+    public class AssemblyHasCommitHashAttributeRule : IPackageVerifierRule
     {
-        public override IEnumerable<PackageVerifierIssue> ValidateAttribute(
-            string currentFilePath,
-            AssemblyDefinition assembly,
-            Collection<CustomAttribute> assemblyAttributes)
+        public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
         {
-            var fileName = Path.GetFileNameWithoutExtension(currentFilePath);
-            var isSourcesPackage = fileName.EndsWith(".Sources", StringComparison.OrdinalIgnoreCase);
-            if (!isSourcesPackage && !HasCommitHashInMetadataAttribute(assemblyAttributes))
+            AssemblyHasAttributeHelper.GetAssemblyAttributesData(context);
```

Comment:
I was actually in the process of moving it to `Program.cs` [here](https://github.com/aspnet/BuildTools/blob/dev/src/NuGetPackageVerifier/Program.cs#L166) and [here](https://github.com/aspnet/BuildTools/blob/dev/src/NuGetPackageVerifier/Program.cs#L229)